keps/sig-network/0015-20180614-SCTP-support.md

```diff
@@ -0,0 +1,258 @@
+---
+kep-number: 15
+title: SCTP support
+authors:
+  - "@janosi"
+owning-sig: sig-network
+participating-sigs:
+  - sig-network
+reviewers:
+  - "@thockin"
+approvers:
+  - "@thockin"
+editor: TBD
+creation-date: 2018-06-14
+last-updated: 2018-06-22
+status: provisional
+see-also:
+  - PR64973
+replaces:
+superseded-by:
+---
+
+# SCTP support
+
+## Table of Contents
+
+* [Table of Contents](#table-of-contents)
+* [Summary](#summary)
+* [Motivation](#motivation)
+    * [Goals](#goals)
+    * [Non-Goals](#non-goals)
+* [Proposal](#proposal)
+    * [User Stories [optional]](#user-stories-optional)
+      * [Story 1](#story-1)
+      * [Story 2](#story-2)
+    * [Implementation Details/Notes/Constraints [optional]](#implementation-detailsnotesconstraints-optional)
+    * [Risks and Mitigations](#risks-and-mitigations)
+* [Graduation Criteria](#graduation-criteria)
+* [Implementation History](#implementation-history)
+* [Drawbacks [optional]](#drawbacks-optional)
+* [Alternatives [optional]](#alternatives-optional)
+
+## Summary
+
+The goal of the SCTP support feature is to enable the usage of the SCTP protocol in Kubernetes [Service][],  [NetworkPolicy][], and [ContainerPort][]as an additional protocol value option beside the current TCP and UDP options.
+SCTP is an IETF protocol specified in [RFC4960][], and it is used widely in telecommunications network stacks.
+Once SCTP support is added as a new protocol option those applications that require SCTP as L4 protocol on their interfaces can be deployed on Kubernetes clusters on a more straightforward way. For example they can use the native kube-dns based service discovery, and their communication can be controlled on the native NetworkPolicy way.
+
+[Service]: https://kubernetes.io/docs/concepts/services-networking/service/
+[NetworkPolicy]: 
+https://kubernetes.io/docs/concepts/services-networking/network-policies/
+[ContainerPort]:https://kubernetes.io/docs/concepts/services-networking/connect-applications-service/#exposing-pods-to-the-cluster
+[RFC4960]: https://tools.ietf.org/html/rfc4960
+
+
+## Motivation
+
+SCTP is a widely used protocol in telecommunications. It would ease the management and execution of telecommunication applications on Kubernetes if SCTP were added as a protocol option to Kubernetes. 
+
+### Goals
+
+Add SCTP support to Kubernetes ContainerPort, Service and NetworkPolicy, so applications running in pods can use the native kube-dns based service discovery for SCTP based services, and their communication can be controlled via the native NetworkPolicy way.
+
+### Non-Goals
+
+It is not a goal here to add SCTP support to load balancers that are provided by cloud providers. 
```

Comment:
I guess I can hold my nose on this.  I really dislike things that are quietly non-portable, like this, but it's hardly the first or worst.

Reply:
Arguably my position on SCTP LoadBalancers here contradicts my position on NetworkPolicy ipBlock ingress on the list... (Although I feel like ipBlock ingress is more broken than SCTP LoadBalancing is so it's not *too* inconsistent.) Having stronger guidelines for situations like this would be good... I think sig-network outsources more behavior to plugins and third parties than most other SIGs do so this is probably something that affects us more than everyone else.